### ISSUE TITLE
Add support for absolute paths in outputs

### ISSUE THREAD

**Issue by ivancuric** [danethurber/webpack-manifest-plugin#87]
I have a use case where I'd like to output the manifest.json in a different folder than where the bundles are output.

However, if I want to output it in the parent directory, I have to use `fileName: '../manifest.json'`.

This is not really maintainable cross-config.

**Comment by mastilver** [comment]
#33 duplicate 

Use an absolute path

**Comment by ivancuric** [comment]
Not working. If using `fileName: path.join(__dirname, 'dist','manifest.json')`, it just gets appended to the output directory, eg:
`/dist/Users/ivan/Dev/projectName/dist/manifest.json`

**Comment by mastilver** [comment]
Weird... We have one test that cover that: https://github.com/danethurber/webpack-manifest-plugin/blob/78c53210716e9c32c9b1d840ab5ce11b4750bbe1/spec/plugin.spec.js#L500-L522

Do you mind sending us a failing test so we can have a better look

**Comment by ivancuric** [comment]
Uh, never done tests before, but I can send you a reduced config:
<details>
 <summary>webpack.config.js</summary>

```js
const ManifestPlugin = require('webpack-manifest-plugin');
const fs = require('fs-extra');
const paths = require('./_scripts/paths');
const manifestPath = paths.dist.root + 'manifest.json';

const basePlugins = () => [
  new ManifestPlugin({
    fileName: manifestPath,
    cache: JSON.parse(fs.readFileSync(manifestPath, 'utf8')),
  }),
];

const config = {
  entry: {
    entry: paths.src.js + 'entry.js',
  },
  output: {
    path: paths.dist.js,
    publicPath: '/static/js/',
    filename: '[name].js',
  },
  module: {
    rules: [
      {
        test: /\.js$/,
        exclude: /(node_modules)/,
        use: {
          loader: 'babel-loader',
          options: {
            cacheDirectory: true,
          },
        },
      },
    ],
  },
  resolve: {
    modules: ['node_modules', paths.src.js],
  },
  plugins: basePlugins(),
};

module.exports = config;
```

</details>

<details>
<summary>scripts/paths.js</summary>

```js
const path = require('path');

// resource paths
const DIR_SRC = path.join(__dirname, '..', 'src');
const DIR_DIST = path.join(__dirname, '..', 'dist');

// prettier-ignore
exports.src = {
  assets  : DIR_SRC + '/assets/',
  html    : DIR_SRC + '/html/',
  js      : DIR_SRC + '/js/',
  markdown: DIR_SRC + '/markdown/',
  styles  : DIR_SRC + '/styles/',
  svg     : DIR_SRC + '/assets/svg/',
  fonts   : DIR_SRC + '/assets/fonts/',
  root    : DIR_SRC + '/',
};

// prettier-ignore
exports.dist = {
  assets  : DIR_DIST + '/assets/',
  html    : DIR_DIST + '/html/',
  js      : DIR_DIST + '/js/',
  markdown: DIR_DIST + '/markdown/',
  styles  : DIR_DIST + '/styles/',
  svg     : DIR_DIST + '/assets/svg/',
  fonts   : DIR_DIST + '/assets/fonts/',
  root    : DIR_DIST + '/',
};
```

</details>


# Output:

![image](https://user-images.githubusercontent.com/2827047/30379245-1c0f5d14-9896-11e7-8560-2b27abe649c2.png)

![image](https://user-images.githubusercontent.com/2827047/30379267-311398ba-9896-11e7-96cf-a2d774ff9c1e.png)

**Comment by mastilver** [comment]
What's your version of `webpack-manifest-plugin`?

**Comment by ivancuric** [comment]
`    "webpack-manifest-plugin": "^1.3.1",`

**Comment by mastilver** [comment]
My bad..., I managed to reproduce it, I will try to have a look tonight

**Comment by mastilver** [comment]
So I find the reason why the tests are passing

The `join` function of https://github.com/webpack/webpack/blob/master/lib/MemoryOutputFileSystem.js and https://github.com/webpack/webpack/blob/master/lib/node/NodeOutputFileSystem.js are behaving defferently

`memoryFs.join('/dir1', '/dir2/file')` => `/dir2/file`
`path.join('/dir1', '/dir2/file')` => `/dir1/dir2/file`

See: https://runkit.com/embed/yp5dtitvt668

**Comment by ivancuric** [comment]
Uh... That's weird.

**Comment by mastilver** [comment]
published `v1.3.2`
